### PR TITLE
switched to fbmessenger library instead of pymessenger

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- remove old pymessenger library and use fbmessenger instead.
- added a couple type annotation and comments here and there

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality (not possible due to FB)
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: rasa_core/channels/facebook.py [hunk 1]
```diff
@@ -5,73 +5,214 @@
...
+                          (can be chosen by yourself on webhook creation)
+        :param fb_secret: facebook application secret
+        :param fb_access_token: access token to post in the name of the FB page
+        """
+        self.fb_verify = fb_verify
+        self.fb_secret = fb_secret
+        self.fb_access_token = fb_access_token
 
     def blueprint(self, on_new_message):
         from pymessenger.utils import validate_hub_signature
```
Comment (amn41) [inline rasa_core/channels/facebook.py]: I guess we wanna get rid of this import

Comment (tmbo) [reply rasa_core/channels/facebook.py]: very good point! good spot.

**Comment by ladak** [comment]
@tmbo, @bastiW - version i am using is this rasa_core-0.8.0a4-py3.6.egg

**Comment by tmbo** [comment]
@ladak ok should be fixed in the latest master version

**Comment by ladak** [comment]
@tmbo - I installed the master version. Still getting the problem.

$ python bot.py run
2018-01-09 17:56:40 INFO     rasa_nlu.components - Added 'nlp_spacy' to component cache. Key 'nlp_spacy-en'.
Using TensorFlow backend.
127.0.0.1 - - [2018-01-09 17:57:10] "GET /app/webhook?hub.mode=subscribe&hub.challenge=1044565230&hub.verify_token=thisisatestbotevahelps HTTP/1.1" 200 126 0.005081
2018-01-09 17:57:33 WARNING  rasa_core.channels.facebook - Wrong fb secret! Make sure this matches the secret in your facebook app settings
127.0.0.1 - - [2018-01-09 17:57:33] "POST /app/webhook HTTP/1.1" 200 129 0.003007

**Comment by ladak** [comment]
@tmbo - Now working with the latest commit.
Thanks.

**Pull request merged**